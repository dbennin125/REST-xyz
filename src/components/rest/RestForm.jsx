import React from 'react';
import PropTypes from 'prop-types';

const RestForm = ({ text }) => (
  <input type="text" name="text" value={text}/>
);

RestForm.propTypes = {
  text: PropTypes.string.isRequired
};

export default RestForm;
