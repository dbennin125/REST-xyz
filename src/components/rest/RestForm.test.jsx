import React from 'react';
import { shallow } from 'enzyme';
import RestForm from './RestForm';


describe('RestForm component', () => {
  let wrapper;
  let handleChange;
  let handleSubmit;
  
  beforeEach(() => {
    handleChange = jest.fn();
    handleSubmit = jest.fn();

    wrapper = shallow(<RestForm 
      URL="Hello World!" 
      method="GET"
      onChange={handleChange}
      onSubmit={handleSubmit}
      body="some String"
    />);
  });

  it('has a text input with a value set to text prop', () => {
    expect(wrapper.find('input[name="url"]').prop('value')).toEqual('Hello World!');
  });
  it('has a GET radio input with a value set to method prop', () => {
    expect(wrapper.find('input[name="method"]').findWhere(n => n.prop('checked')).prop('value')).toEqual('GET');
  });
  it('invokes an onChange prop when text input is changed', () => {
    wrapper.find('input[name="url"]').simulate('change');
    expect(handleChange).toHaveBeenCalledTimes(1);
  });
  it('invokes an onChange prop when radio input is changed', () => {
    wrapper.find('input[name="method"]').findWhere(radios => radios.prop('checked')).simulate('change');
    expect(handleChange).toHaveBeenCalledTimes(1);
  });
  it('invokes onSubmit prop for the form when button pressed', () => {
    wrapper.find('form').simulate('submit');
    expect(handleSubmit).toHaveBeenCalledTimes(1);
  });

  it('renders the component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
  
