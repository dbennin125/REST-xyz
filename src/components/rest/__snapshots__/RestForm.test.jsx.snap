// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RestForm component renders the component 1`] = `
<form
  onSubmit={[MockFunction]}
>
  <input
    name="url"
    onChange={[MockFunction]}
    type="text"
    value="Hello World!"
  />
  <label>
    <input
      checked={true}
      name="method"
      onChange={[MockFunction]}
      type="radio"
      value="GET"
    />
    GET
  </label>
  <label>
    <input
      checked={false}
      name="method"
      onChange={[MockFunction]}
      type="radio"
      value="PUT"
    />
    PUT
  </label>
  <label>
    <input
      checked={false}
      name="method"
      onChange={[MockFunction]}
      type="radio"
      value="POST"
    />
    POST
  </label>
  <label>
    <input
      checked={false}
      name="method"
      onChange={[MockFunction]}
      type="radio"
      value="DELETE"
    />
    DELETE
  </label>
  <button>
    Submit button
  </button>
  <textarea
    name="body"
    onChange={[MockFunction]}
    value="some String"
  />
</form>
`;
